app/routes.py

```
@@ -101,88 +106,198 @@ async def join_meeting(request: BotRequest, client_request: Request):
+    resolved_persona_data: Dict[str, Any] = {}
+    persona_name_for_logging: str = "Unknown"
+
+    if request.prompt: # Case 1: Custom prompt provided (dynamic persona)
+        logger.info(f"Processing custom prompt for bot {bot_client_id}")
+        prompt_derived_details = await extract_persona_details_from_prompt(request.prompt)
+
+        if prompt_derived_details and isinstance(prompt_derived_details, dict): # Ensure it's a dict
+            # Construct resolved_persona_data from derived details
+            resolved_persona_data = {
+                "name": prompt_derived_details.get("name", "Bot"),
+                "prompt": request.prompt, # Store original request prompt as the base prompt for dynamic persona
```

Pylint (3.3.7)

[convention] Line too long (112/100)

(C0301)